lib/mock/mock-agent.js

```diff
@@ -0,0 +1,111 @@
+'use strict'
+
+const { kAgentOpts, kAgentCache } = require('../core/symbols')
+const { Agent } = require('../agent')
+const {
+  kMockAgentSet,
+  kMockAgentGet,
+  kClose,
+  kOriginalClose,
+  kDispatches,
+  kCreatePool,
+  kIsMockActive,
+  kNetConnect,
+  kGetNetConnect
+} = require('./mock-symbols')
+const MockClient = require('./mock-client')
+const MockPool = require('./mock-pool')
+const { matchValue } = require('./mock-utils')
+const { InvalidArgumentError } = require('../core/errors')
+
+class MockAgent extends Agent {
```

Comment:
Looking at https://github.com/nodejs/undici/pull/600 it might be better that the MockAgent encapsulates an Agent rather than inherit from it.

Reply:
No problem, will get right on it!